packages/common/src/queue.test.ts

```diff
@@ -0,0 +1,349 @@
+import { assertType, expect, test, vi } from "vitest";
+import { promiseWithResolvers } from "./promiseWithResolvers.js";
+import { createQueue } from "./queue.js";
+
+test("add resolves", async () => {
+  const queue = createQueue({
+    concurrency: 1,
+    frequency: 1,
+    worker: () => Promise.resolve(1),
+  });
+
+  queue.start();
+
+  const promise = queue.add();
+
+  expect(await promise).toBe(1);
+});
+
+test("add rejects", async () => {
+  let rejected = false;
+
+  const queue = createQueue({
+    concurrency: 1,
+    frequency: 1,
+    worker: () => Promise.reject(),
+  });
+
+  const promise = queue.add();
+
+  queue.start();
+
+  await promise.catch(() => {
+    rejected = true;
+  });
+
+  expect(rejected).toBe(true);
+});
+
+test("size", () => {
+  const queue = createQueue({
+    concurrency: 1,
+    frequency: 1,
+    worker: () => Promise.resolve(),
+  });
+
+  queue.add();
+
+  expect(queue.size()).toBe(1);
+
+  queue.start();
+
+  expect(queue.size()).toBe(0);
+});
+
+test("pending", async () => {
+  const { promise, resolve } = promiseWithResolvers<void>();
+
+  const queue = createQueue({
+    concurrency: 1,
+    frequency: 1,
+    worker: () => promise,
+  });
+
+  queue.start();
+
+  queue.add();
+
+  expect(await queue.pending()).toBe(1);
+
+  resolve();
+
+  expect(await queue.pending()).toBe(0);
+});
+
+test("clear", () => {
+  const queue = createQueue({
+    concurrency: 1,
+    frequency: 1,
+    worker: () => Promise.resolve(),
+  });
+
+  queue.add();
+  queue.add();
+  queue.add();
+
+  queue.clear();
+
+  expect(queue.size()).toBe(0);
+});
+
+test("clear timer", async () => {
+  const queue = createQueue({
+    frequency: 1,
+    worker: () => Promise.resolve(),
+  });
+
+  queue.add();
+  queue.add();
+  queue.add();
+
+  queue.start();
+  queue.clear();
+
+  await queue.onIdle();
+
+  expect(queue.size()).toBe(0);
+  expect(await queue.pending()).toBe(0);
+});
+
+test("isStarted", () => {
+  const queue = createQueue({
+    concurrency: 1,
+    frequency: 1,
+    worker: () => Promise.resolve(),
+  });
+
+  expect(queue.isStarted()).toBe(false);
+
+  queue.start();
+
+  expect(queue.isStarted()).toBe(true);
+});
+
+test("start", async () => {
+  const queue = createQueue({
+    concurrency: 1,
+    frequency: 1,
+    worker: () => Promise.resolve(),
+  });
+
+  const promise = queue.add();
+
+  expect(queue.size()).toBe(1);
+
+  queue.start();
+
+  expect(queue.isStarted()).toBe(true);
+
+  await promise;
+
+  expect(queue.size()).toBe(0);
+});
+
+test("pause", () => {
+  const queue = createQueue({
+    concurrency: 1,
+    frequency: 1,
+    worker: () => Promise.resolve(),
+  });
+
+  queue.start();
+  queue.pause();
+
+  queue.add();
+
+  expect(queue.size()).toBe(1);
+});
+
+test("onIdle short loop", async () => {
+  const queue = createQueue({
+    concurrency: 1,
+    frequency: 1,
+    worker: () => Promise.resolve(),
+  });
+
+  await queue.onIdle();
```

Comment:
No `expect` here? Also a few other tests in this file

Reply:
These tests are testing that the promise resolves. If the test were to fail, it would basically just timeout